crates/yas_core/src/tests/test_contracts/test_yas_pool.cairo

```diff
@@ -486,81 +491,1014 @@ mod YASPoolTests {
-        // TODO: 'initialize the pool at price of 10:1'
+        use yas_core::utils::math_utils::{
+            FullMath::{div_rounding_up, mul_div, mul_div_rounding_up}, pow
+        };
+
+
+        #[test]
+        #[available_gas(2000000000)]
+        #[should_panic(expected: ('LOK', 'ENTRYPOINT_FAILED'))]
+        fn test_fails_not_initialized() {
+            let yas_pool = deploy(
+                FACTORY_ADDRESS(),
+                TOKEN_A(),
+                TOKEN_B(),
+                fee_amount(FeeAmount::MEDIUM),
+                IntegerTrait::<i32>::new(tick_spacing(FeeAmount::MEDIUM), false)
+            );
+            let sqrt_price_X96 = encode_price_sqrt_1_1();
+            //yas_pool.initialize(sqrt_price_X96); //don't
+            let (min_tick, max_tick) = get_min_tick_and_max_tick();
+
+            let (amount0, amount1): (u256, u256) = yas_pool
+                .mint(
+                    recipient: yas_pool.contract_address,
+                    tick_lower: min_tick,
+                    tick_upper: max_tick,
+                    amount: 1,
+                    data: ArrayTrait::<felt252>::new()
+                );
+        }
+
 
         mod FailureCases {
-            // TODO: 'fails if tickLower greater than tickUpper'
+            use super::{
+                setup, MIN_TICK, MAX_TICK, tick_spacing, FeeAmount, fee_amount,
+                IERC20DispatcherTrait, FACTORY_ADDRESS, TOKEN_A, TOKEN_B, WALLET,
+                encode_price_sqrt_1_1
+            };
+            use yas_core::numbers::signed_integer::{
+                i32::i32, i32::i32_div_no_round, integer_trait::IntegerTrait
+            };
+            use yas_core::contracts::yas_pool::{
+                YASPool, YASPool::ContractState, YASPool::YASPoolImpl, YASPool::InternalImpl,
+                IYASPool, IYASPoolDispatcher, IYASPoolDispatcherTrait
+            };
+            use yas_core::contracts::yas_router::{
+                YASRouter, IYASRouterDispatcher, IYASRouterDispatcherTrait
+            };
+
+
+            #[test]
+            #[available_gas(2000000000)]
+            #[should_panic(expected: ('TLU', 'ENTRYPOINT_FAILED', 'ENTRYPOINT_FAILED'))]
+            fn test_fails_tick_lower_greater_than_tick_upper() {
+                let (yas_pool, token_0, token_1, mint_callback, min_tick, max_tick) = setup();
+                mint_callback
+                    .mint(
+                        yas_pool.contract_address,
+                        WALLET(),
+                        IntegerTrait::<i32>::new(1, false),
+                        IntegerTrait::<i32>::new(0, false),
+                        1
+                    );
+            }
+
+
+            #[test]
+            #[available_gas(2000000000)]
+            #[should_panic(expected: ('TLM', 'ENTRYPOINT_FAILED', 'ENTRYPOINT_FAILED'))]
+            fn test_fails_tick_lower_than_min() {
+                let (yas_pool, token_0, token_1, mint_callback, min_tick, max_tick) = setup();
```

Comment:
same here, replace unused variables from setup() with `_`.

Reply:
Also check this in other tests :pray:

Reply:
done, this made the code much cleaner, thanks!